src/BuildScriptGeneratorCli/ServiceProviderBuilder.cs

```diff
@@ -72,6 +72,18 @@ private static LoggingConfiguration BuildNLogConfiguration([CanBeNull] string lo
 
             var aiKey = disableTelemetry ? string.Empty : Environment.GetEnvironmentVariable(
                 LoggingConstants.ApplicationInsightsInstrumentationKeyEnvironmentVariableName);
+            var aiConnectionString = disableTelemetry ? string.Empty : Environment.GetEnvironmentVariable(LoggingConstants.ApplicationInsightsConnectionStringKeyEnvironmentVariableName);
+            if (!string.IsNullOrEmpty(aiConnectionString))
+            {
+                var aiTarget = new ApplicationInsights.NLogTarget.ApplicationInsightsTarget()
+                {
+                    Name = "ai",
+                    InstrumentationKey = aiConnectionString,
+                };
+                config.AddTarget(aiTarget);
+                config.AddRuleForAllLevels(aiTarget);
+            }
+
             if (!string.IsNullOrWhiteSpace(aiKey))
```

Comment:
We should make this an `else if` so that we only use the instrumentation key when the connection string isn't provided